### PR TITLE
custom names for arrays

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by burrbull** [comment]
cc @Emilgardis

File: svd-rs/src/array.rs [hunk 1]
```diff
@@ -50,11 +50,51 @@ where
...
+        dim.dim_array_index
+            .as_ref()
+            .map(|dai| {
+                dai.values
+                    .iter()
+                    .find(|e| e.value.map(|v| v.to_string().as_str() == i.deref()) == Some(true))
+            })
+            .flatten()
+            .map(|n| n.description.clone())
+            .flatten()
```
Comment (Emilgardis) [inline svd-rs/src/array.rs:95]: `Some(_).map(...).flatten()` is the same as `and_then`

File: svd-rs/src/array.rs [hunk 2]
```diff
@@ -50,11 +50,51 @@ where
...
-        .map(move |i| name.replace("[%s]", &i).replace("%s", &i))
+    dim.indexes().map(move |i| {
+        dim.dim_array_index
+            .as_ref()
+            .map(|dai| {
+                dai.values
+                    .iter()
+                    .find(|e| e.value.map(|v| v.to_string().as_str() == i.deref()) == Some(true))
+            })
+            .flatten()
```
Comment (Emilgardis) [inline svd-rs/src/array.rs:73]: `Some(_).map(...).flatten()` is the same as `and_then`

**Review by Emilgardis — CHANGES_REQUESTED** [review]

File: svd-rs/src/array.rs (hunk 1 above)
Comment (burrbull) [reply svd-rs/src/array.rs]: Exactly. Thanks.

**Review by Emilgardis — APPROVED** [review]

**Pull request merged**